src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FeePaymentHandler.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Arrays;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacaseapi.domain.service.FeePayment;
+
+@Component
+public class FeePaymentHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private final FeePayment<AsylumCase> feePayment;
+
+    public FeePaymentHandler(
+            FeePayment<AsylumCase> feePayment
+    ) {
+        this.feePayment = feePayment;
+    }
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+               &&
+               Arrays.asList(
+                   Event.SUBMIT_APPEAL
+               ).contains(callback.getEvent());
```

Comment:
To enable us to deploy your changes here in isolation it may be a good idea to only trigger the fees/payments handler when a feature flag is turned on.  That way we can turn it off and merge these changes without relying on another service being there.

There is an existing pattern to do this with Launch Darkly ( programme has a license to the product ) which should make it quite simple.  You will however have to provide functional tests for both sides of the flag.

Reply:
Added feature flag with Launch Darkly. Functional tests added to cover both sides of the flag.